File: JavaAndScala/src/main/java/com/java/se/conclusion/design/factory/simple/common/laptop/Acer.java
package com.java.se.conclusion.design.factory.simple.common.laptop;

/**
 * 	This is a class to be used to test the implementation of simple factory pattern
 *  -- This class must implement a parent interface
 *  
 * @author VinceYuan
 *
 */
public class Acer implements Laptop {

	@Override
	public void startup() {
		System.out.println("An Acer laptop is starting up...");
	}

	@Override
	public void shutdown() {
		System.out.println("An Acer laptop is shutting down...");
	}
}

File: JavaAndScala/src/main/scala/org/lang/scala/conclusion/design/factory/simple/common/laptop/Acer.scala
package org.lang.scala.conclusion.design.factory.simple.common.laptop

import org.lang.scala.conclusion.design.factory.simple.common.laptop.parent.Laptop

/**
 * 	This is a stand-alone class to be used to test the implementation of simple factory pattern
 *  -- This class must implement a parent interface
 *  
 * 	@author VinceYuan
 */
class Acer extends Laptop {
  
  override def startup(): Unit = {
    println("An Acer laptop is starting up...")
  }
  
  override def shutdown(): Unit = {
    println("An Acer laptop is shutting down...")
  }
}